### PR TITLE
podvm_mkosi: multistage binaries

### PR DESCRIPTION
**Pull request by stevenhorsman**
Simplify the podvm_mkosi binaries building process by combining the builder and binaries into a single multistage dockerfile

### CODE REVIEW

**Comment by stevenhorsman** [comment]
This has been tested on x86 in my fork: https://github.com/stevenhorsman/cloud-api-adaptor/actions/runs/12256747305

**Review by mkulke — APPROVED** [review]

File: src/cloud-api-adaptor/podvm-mkosi/Makefile [hunk 1]
```diff
@@ -64,8 +54,9 @@ endif
 		--build-arg VERIFY_PROVENANCE=$(VERIFY_PROVENANCE) \
 		$(if $(AUTHFILE),--build-arg AUTHFILE=$(AUTHFILE),) \
 		$(if $(DEFAULT_AGENT_POLICY_FILE),--build-arg DEFAULT_AGENT_POLICY_FILE=$(DEFAULT_AGENT_POLICY_FILE),) \
-		-o type=local,dest="./resources/binaries-tree" \
-		-f ../podvm/Dockerfile.podvm_binaries.fedora ../../
+		$(if $(filter $(PUSH),true),,-o type=local,dest="./resources/binaries-tree") \
+		$(DOCKER_OPTS) \
+		-f ../podvm/Dockerfile.podvm.fedora ../../
```
Comment (wainersm) [inline src/cloud-api-adaptor/podvm-mkosi/Makefile:59]: Hi @stevenhorsman !

Shouldn't it still use `Dockerfile.podvm_binaries.fedora`?

File: src/cloud-api-adaptor/podvm/Dockerfile.podvm_binaries.fedora [hunk 2]
```diff
@@ -1,12 +1,57 @@
...
-# Builds pod vm image inside container
+# Build binaries for mkosi podvm image
 #
-ARG BUILDER_IMG
+FROM registry.fedoraproject.org/fedora:40 AS builder
 
-FROM ${BUILDER_IMG} AS podvm_builder
+ARG ARCH="amd64"
+ARG YQ_ARCH="amd64"
+# PROTOC_ARCH="x86_64" | "s390_64"
```
Comment (wainersm) [inline src/cloud-api-adaptor/podvm/Dockerfile.podvm_binaries.fedora:12]: drop that line.

File: src/cloud-api-adaptor/podvm-mkosi/Makefile (hunk 1 above)
Comment (stevenhorsman) [reply src/cloud-api-adaptor/podvm-mkosi/Makefile:59]: Sorry, I renamed this last minute (I created a new one in order to keep the old code untouched before deciding this didn't make sense). I will fix and re-test. thanks for the spot!

**Pull request merged**